src/rdkafka_admin.c

```diff
@@ -4801,3 +5055,1608 @@ void rd_kafka_DeleteAcls(rd_kafka_t *rk,
+    const rd_kafka_MemberDescription_t *member) {
+        rd_assert(member != NULL);
+        return member->client_id;
+}
+
+const char *rd_kafka_MemberDescription_group_instance_id(
+    const rd_kafka_MemberDescription_t *member) {
+        rd_assert(member != NULL);
+        return member->group_instance_id;
+}
+
+const char *rd_kafka_MemberDescription_consumer_id(
+    const rd_kafka_MemberDescription_t *member) {
+        rd_assert(member != NULL);
+        return member->consumer_id;
+}
+
+const char *
+rd_kafka_MemberDescription_host(const rd_kafka_MemberDescription_t *member) {
+        rd_assert(member != NULL);
+        return member->host;
+}
+
+const rd_kafka_MemberAssignment_t *rd_kafka_MemberDescription_assignment(
+    const rd_kafka_MemberDescription_t *member) {
+        rd_assert(member != NULL);
+        return &member->assignment;
+}
+
+const rd_kafka_topic_partition_list_t *
+rd_kafka_MemberAssignment_topic_partitions(
+    const rd_kafka_MemberAssignment_t *assignment) {
+        rd_assert(assignment != NULL);
+        return assignment->topic_partitions;
+}
+
+
+/**
+ * @brief Create a new ConsumerGroupDescription object.
+ *
+ * @param group_id The group id.
+ * @param is_simple_consumer_group Is the group simple?
+ * @param members List of members (rd_kafka_MemberDescription_t) of this
+ *                group.
+ * @param partition_assignor (optional) Chosen assignor.
+ * @param state Group state.
+ * @param coordinator (optional) Group coordinator.
+ * @param error (optional) Error received for this group.
+ * @return A new allocated ConsumerGroupDescription object.
+ *         Use rd_kafka_ConsumerGroupDescription_destroy() to free when done.
+ */
+static rd_kafka_ConsumerGroupDescription_t *
+rd_kafka_ConsumerGroupDescription_new(const char *group_id,
+                                      rd_bool_t is_simple_consumer_group,
+                                      const rd_list_t *members,
+                                      const char *partition_assignor,
+                                      rd_kafka_consumer_group_state_t state,
+                                      const rd_kafka_Node_t *coordinator,
+                                      rd_kafka_error_t *error) {
+        rd_kafka_ConsumerGroupDescription_t *grpdesc;
+        grpdesc                           = rd_calloc(1, sizeof(*grpdesc));
+        grpdesc->group_id                 = rd_strdup(group_id);
+        grpdesc->is_simple_consumer_group = is_simple_consumer_group;
+        if (members == NULL) {
+                rd_list_init(&grpdesc->members, 0,
+                             rd_kafka_MemberDescription_free);
+        } else {
+                rd_list_init_copy(&grpdesc->members, members);
+                rd_list_copy_to(&grpdesc->members, members,
+                                rd_kafka_MemberDescription_list_copy, NULL);
+        }
+        grpdesc->partition_assignor = !partition_assignor
```

Comment:
This reads a bit weird, be more explicit with your intention in this case, so:
`partition_assignor = partition_assignor ? rd_strdup(partition_assignor) : NULL;`

Reply:
But you don't really need the NULL assign since the struct is calloced, so do an if instead.